variant/JsonStrings.js

```diff
@@ -0,0 +1,77 @@
+// The parseParameter function will typically be called for every page and 
+// the setParameter function is called by each active plugin to bundle specific attacks
+
+// Note that new custom input vector scripts will initially be disabled
+// Right click the script in the Scripts tree and select "enable"  
+
+// The following handles differences in printing between Java 7's Rhino JS engine
+// and Java 8's Nashorn JS engine
+if (typeof println == 'undefined') this.println = print;
+
+/*
+This variant script can be used as an alternative to the default JSON input vectors
+when one only wants to scan the string fields of the JSON object, not integers.
+
+Input vectors will be located and set recursively and objects containing up to three
+layers of objects, lists and strings seem to work as intended.
+*/
+
+function parseParameters(helper, msg) {
+	//we're only interested in non-empty JSON requests
+	header = msg.getRequestHeader()
+	if(header.getHeader("Content-Type") && !header.getHeader("Content-Type").trim().equalsIgnoreCase("application/json")
```

Comment:
@juhakivekas would you be available to take a look at this? Otherwise we can tweak it if not.